packages/main/src/plugin/dockerode/libpod-dockerode.ts

```diff
@@ -206,6 +207,44 @@ export class LibpodDockerode {
       });
     };
 
+    // add attach
+    prototypeOfDockerode.podmanAttach = function (containerId: string) {
+      const optsf = {
+        path: `/v4.2.0/libpod/containers/${containerId}/attach?stdin=true&stdout=true&stderr=true&`,
+        method: 'POST',
+        isStream: true,
+        openStdin: true,
+        allowEmpty: true,
+        statusCodes: {
+          200: true,
+          404: 'no such container',
+          500: 'server error',
+        },
+        options: {},
+      };
+
+      // patch the modem to not send any data. By default dockerode send query parameters as JSON payload
+      // but podman REST API will then echo the response, so send empty data '' instead
+      const originalBuildRequest = this.modem.buildRequest;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      this.modem.buildRequest = function (options: unknown, context: any, data: unknown, callback: unknown) {
+        if (context.allowEmpty && context.path.includes('/attach?')) {
+          data = '';
+        }
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        return originalBuildRequest.call(this, options, context, data, callback);
+      };
+
+      return new Promise((resolve, reject) => {
+        this.modem.dial(optsf, (err: unknown, stream: unknown) => {
+          if (err) {
+            return reject(err);
+          }
+          resolve(stream);
+        });
+      });
```

Comment:
```suggestion
      return dial(optsf);
```

Reply:
and down below where applicable to avoid code duplication.

Reply:
it looks like a refactoring as it applies to all the other methods, so I'll handle in a follow-up PR.

for now it was reusing the same approach than the dockerode library 
 https://github.com/apocas/dockerode/blob/master/lib/docker.js with duplication.